### PR TITLE
remove init_context

### PR DESCRIPTION
**Pull request by LielinJiang**
1. remove init_context
2. init multiple in model __init__
3. set place again in model __init__
4. remove get_local_rank and get_nranks, use Env() instead
5. remove some noused import
6. remove DistributedDataParallel

### CODE REVIEW

File: model.py [hunk 1]
```diff
@@ -721,8 +705,19 @@ def __init__(self):
         self._test_dataloader = None
 
         # init multiple gpus context
-        self._place = fluid.CUDAPlace(distributed.Env().dev_id) \
-                    if distributed.Env().nranks > 1 else fluid.CUDAPlace(0)
+        self._place = fluid.CUDAPlace(ParallelEnv().dev_id) \
```
Comment (guoshengCS) [inline model.py]: Should we add CPUPlace here
Additionally should we need a way to switch between CPU and GPU

File: model.py [hunk 2]
```diff
@@ -1035,7 +1046,7 @@ def _run_one_epoch(data_loader, callbacks, mode):
                 loader = eval_loader
                 if not isinstance(eval_loader, Iterable):
                     loader = eval_loader()
-                logs = _run_one_epoch(eval_loader(), cbks, 'eval')
+                logs = _run_one_epoch(eval_loader, cbks, 'eval')
```
Comment (guoshengCS) [inline model.py]: Should `eval_loader` be `loader`

File: model.py (hunk 1 above)
Comment (LielinJiang) [reply model.py]: updated, thanks.

File: model.py (hunk 2 above)
Comment (LielinJiang) [reply model.py]: done, thanks.

File: distributed.py [hunk 3]
```diff
@@ -106,27 +98,22 @@ def __len__(self):
         num_samples += int(not self.drop_last) * (self.batch_size - 1)
         return num_samples // self.batch_size
 
+    def set_epoch(self, epoch):
```
Comment (qingqing01) [inline distributed.py]: 这个函数没看到其他地方用

File: model.py [hunk 4]
```diff
@@ -474,31 +469,28 @@ def _compile_and_initialize(self, prog, mode):
...
 
-        device = self.model._device
-        device_ids = self.model._device_ids
-
-        if device.lower() == 'gpu':
-            places = fluid.cuda_places(device_ids)
-        else:
-            places = fluid.cpu_places(len(device_ids) if device_ids else None)
+        assert self.model._place is not None, \
+            "model not ready, please call `model.prepare()` first"
```
Comment (qingqing01) [inline model.py]: 报错信息优化下：
device is not set, please set it in `model.prepare()` first

File: model.py [hunk 5]
```diff
@@ -474,31 +469,28 @@ def _compile_and_initialize(self, prog, mode):
...
-                place = fluid.CUDAPlace(gpu_id) if device.lower(
-                ) == 'gpu' else fluid.CPUPlace()
-            else:
-                place = places[0]
+            # if self._nranks > 1 and device.lower() == 'gpu':
+            #     gpu_id = int(ParallelEnv().dev_id)
+            #     place = fluid.CUDAPlace(gpu_id) if device.lower(
+            #     ) == 'gpu' else fluid.CPUPlace()
+            # else:
+            #     place = places[0]
```
Comment (qingqing01) [inline model.py]: remove

File: distributed.py (hunk 3 above)
Comment (LielinJiang) [reply distributed.py]: 这里没用到，要是用户想要定义自己的seed，需要给他这个接口。

File: model.py (hunk 4 above)
Comment (LielinJiang) [reply model.py]: done, thanks

File: model.py (hunk 5 above)
Comment (LielinJiang) [reply model.py]: done, thanks

**Pull request merged**